backend/src/main/java/ca/mcgill/purposeful/controller/IdeaController.java

```diff
@@ -49,12 +61,49 @@ public ResponseEntity<IdeaDTO> getIdeaById(@PathVariable String id) {
   @PreAuthorize("hasAnyAuthority('User', 'Moderator', 'Owner')")
   public ResponseEntity<List<IdeaDTO>> filterIdeas(@RequestBody SearchFilterDTO searchFilterDTO) {
     return ResponseEntity.status(HttpStatus.OK)
-        .body(
-            IdeaDTO.convertToDto(
-                ideaService.getIdeasByAllCriteria(
-                    searchFilterDTO.getDomains(),
-                    searchFilterDTO.getTopics(),
-                    searchFilterDTO.getTechnologies())));
+        .body(IdeaDTO.convertToDto(ideaService.getIdeasByAllCriteria(searchFilterDTO.getDomains(),
+            searchFilterDTO.getTopics(), searchFilterDTO.getTechnologies())));
+  }
+
+  /**
+   * This method creates an idea
+   *
+   * @return created idea
+   * @throws Exception
+   * @author Adam Kazma
+   */
+  @PostMapping(value = {"/create", "/create/"})
+  @PreAuthorize("hasAuthority('User')")
+  public IdeaDTO createIdea(@RequestBody IdeaDTO ideaDto) throws Exception {
+    Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+    if (auth == null) {
+      throw new Exception("User not authenticated.");
+    }
+    RegularUser user = regularUserRepository.findRegularUserByAppUserEmail(auth.getName());
```

Comment:
Here we should ideally avoid using the repository. Instead, you can pass in the argument of the email to your service and get the user from within the service